pkg/operator/set_operator_split.go

```diff
@@ -0,0 +1,143 @@
+package operator
+
+import (
+	"sort"
+
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/internal/common/flags"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/operator/split"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/rewards"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/telemetry"
+	"github.com/Layr-Labs/eigenlayer-cli/pkg/utils"
+	"github.com/Layr-Labs/eigensdk-go/chainio/clients/elcontracts"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	eigenSdkUtils "github.com/Layr-Labs/eigensdk-go/utils"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/urfave/cli/v2"
+)
+
+func SetOperatorSplitCmd(p utils.Prompter) *cli.Command {
+	var operatorSplitCmd = &cli.Command{
+		Name:  "set-rewards-split",
+		Usage: "Set operator rewards split",
+		Action: func(cCtx *cli.Context) error {
+			return SetOperatorSplit(cCtx, p)
+		},
+		After: telemetry.AfterRunAction(),
+		Flags: getOperatorSplitFlags(),
+	}
+
+	return operatorSplitCmd
+}
+
+func SetOperatorSplit(cCtx *cli.Context, p utils.Prompter) error {
+	ctx := cCtx.Context
+	logger := common.GetLogger(cCtx)
+
+	config, err := readAndValidateSetOperatorSplitConfig(cCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate operator split config", err)
+	}
+
+	cCtx.App.Metadata["network"] = config.ChainID.String()
+
+	ethClient, err := ethclient.Dial(config.RPCUrl)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to create new eth client", err)
+	}
+
+	eLWriter, err := common.GetELWriter(
+		config.OperatorAddress,
+		config.SignerConfig,
+		ethClient,
+		elcontracts.Config{
+			RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
+		},
+		p,
+		config.ChainID,
+		logger,
+	)
+
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to get EL writer", err)
+	}
+
+	logger.Infof("Broadcasting set operator transaction...")
+
+	var receipt *types.Receipt
+
+	receipt, err = eLWriter.SetOperatorAVSSplit(ctx, config.OperatorAddress, config.AVSAddress, config.Split, true)
+
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to process claim", err)
+	}
+
+	logger.Infof("Set operator transaction submitted successfully")
+	common.PrintTransactionInfo(receipt.TxHash.String(), config.ChainID)
+
+	return nil
+}
+
+func getOperatorSplitFlags() []cli.Flag {
+	baseFlags := []cli.Flag{
+		&flags.NetworkFlag,
```

Comment:
this needs a `--broadcast`  , output type and output file flag. check rewards claim command.

Reply:
how would that be useful for something simple like setting split? Is it to check what would be executed?

Reply:
we caught up on slack, this is needed because users may want to use their own signing solutions. I pushed support for the flags based on rewards claiming, we will want to refactor later to make the broadcast / output code more reusable across commands